src/server/cmd/job-shim/main.go

```
@@ -30,6 +38,74 @@ func main() {
 	env.Main(do, &appEnv{})
 }
 
+func downloadInput(c *client.APIClient, commitMounts []*fuse.CommitMount) error {
+	for _, commitMount := range commitMounts {
+		if commitMount.Alias == "prev" || commitMount.Alias == "out" {
+			continue
+		}
+		repo := commitMount.Commit.Repo.Name
+		commitID := commitMount.Commit.ID
+		var fromCommitID string
+		if commitMount.DiffMethod != nil && commitMount.DiffMethod.FromCommit != nil {
+			fromCommitID = commitMount.DiffMethod.FromCommit.ID
+		}
+		var fullFile bool
+		if commitMount.DiffMethod != nil {
+			fullFile = commitMount.DiffMethod.FullFile
+		}
+		shard := commitMount.Shard
+		fileInfos, err := c.ListFile(repo, commitID, "/", fromCommitID, fullFile, shard, false)
+		if err != nil {
+			return err
+		}
+		for _, fileInfo := range fileInfos {
+			if err := os.MkdirAll(filepath.Join(PFSInputPrefix, repo), 0777); err != nil {
+				return err
+			}
+			path := filepath.Join(PFSInputPrefix, repo, fileInfo.File.Path)
+			f, err := os.Create(path)
+			if err != nil {
+				return err
+			}
+			defer f.Close()
+			w := bufio.NewWriter(f)
+			if err := c.GetFile(repo, commitID, fileInfo.File.Path, 0, 0, fromCommitID, fullFile, shard, w); err != nil {
+				return err
+			}
+			if err := w.Flush(); err != nil {
+				return err
+			}
+		}
+	}
+	return os.MkdirAll(PFSOutputPrefix, 0777)
+}
+
+func uploadOutput(c *client.APIClient, out *fuse.CommitMount, overwrite bool) error {
+	repo := out.Commit.Repo.Name
+	commit := out.Commit.ID
+	return filepath.Walk(PFSOutputPrefix, func(path string, info os.FileInfo, err error) error {
+		if path == PFSOutputPrefix {
+			return nil
+		}
+		f, err := os.Open(path)
+		if err != nil {
+			return err
+		}
+		defer f.Close()
```

Like above, don't swallow this error.